File: TicketToRideUnity/Assets/Scripts/AI/Case.cs
using System;
using System.Runtime.Serialization;

namespace Assets.Scripts.AI
{
    [Serializable]
    public class Case
    {
        public string plan;
        public Situation situation;
        public string caseType;

        public Case()
        {

        }

        public Case(string plan, Situation situation, string caseType)
        {
            this.plan = plan;
            this.situation = situation;
            this.caseType = caseType;
        }
        
        public Case(string plan, string caseType)
        {
            this.plan = plan;
            this.caseType = caseType;
        }

        public Case(string plan, Situation situation)
        {
            this.plan = plan;
            this.situation = situation;
        }
    }
}

File: TicketToRideUnity/Java/TicketToRide/TicketToRide_CBR_System/src/model/Case.java
package model;


public class Case {
	
	public String plan;
	public Situation situation;
	public String caseType;

	public Case() {

	}

	public Case(String plan, Situation situation, String caseType) {
        this.plan = plan;
        this.situation = situation;
        this.caseType = caseType;
	}
	
	public Case(String plan, String caseType) {
		this.plan = plan;
		this.caseType = caseType;
	}

}
